source/documentation/breaking-changes/legacy-js-api.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Breaking Change: Legacy JS API"
+introduction: |
+  Dart Sass originally used an API based on the one used by Node Sass, but
+  replaced it with a new, modern API in Dart Sass 1.45.0. The legacy JS API is
+  now deprecated and will be removed in Dart Sass 2.0.0.
+---
+
+## Migrating Usage
+
+### Entrypoints
+
+The legacy JS API had two entrypoints for compiling Sass: `render` and
+`renderSync`, which took in an options object that included either `file` (to
+compile a file) or `data` (to compile a string). The modern API has four:
+`compile` and `compileAsync` for compiling a file and `compileString` and
+`compileStringAsync` for compiling a string. These functions take a path or
+source string as their first argument and then an object of all other options
+as their second argument. Unlike `render`, which used a callback, `compileAsync`
+and `compileStringAsync` return a promise instead.
+
+See the [usage documentation] for more details.
+
+[usage documentation]: /documentation/js-api/#md:usage
+
+### Importers
+
+Importers in the legacy API consisted of a single function that took in the
+dependency rule URL and the URL of the containing stylesheet (as well as a
+`done` callback for asynchronous importers) and return an object with either
+a `file` path on disk or the `contents` of the stylesheet to be loaded.
+
+Modern API [`Importer`]s instead contain two methods: `canonicalize`, which takes
+in a rule URL and returns the canonical form of that URL; and `load`, which
+takes in a canonical URL and returns an object with the contents
+of the loaded stylesheet. Asynchronous importers have both of these methods
+return promises.
+
+There's also a special [`FileImporter`] that redirects all loads to existing
+files on disk, which should be used when migrating from legacy importers that
+returned a `file` instead of `contents`.
+
+[`Importer`]: /documentation/js-api/interfaces/importer/
+[`ImporterResult`]: /documentation/js-api/interfaces/importerresult/
+[`FileImporter`]: /documentation/js-api/interfaces/fileimporter/
+
+### Custom Functions
+
+In the legacy JS API, custom functions took a separate JS argument for each
+Sass argument, with an additional `done` callback for asynchronous custom
+functions. In the modern API, custom functions instead take a single JS argument
+that contains a list of all Sass arguments, with asynchronous custom functions
+returning a promise.
+
+The modern API also uses a much more robust [`Value`] class that supports
+type assertions and easy map and list lookups.
```

Comment:
I'd add "all Sass value types" to the list of things the new function API supports

Reply:
Done